packages/expo-splash-screen/README.md

```diff
@@ -187,6 +187,45 @@ const styles = StyleSheet.create({
 });
 ```
 
+### Example with hooks: Use `SplashScreen` to replace the `<AppLoading />` component
+
+`App.tsx`
+```tsx
+import React, { useState, useEffect } from 'react';
+import * as SplashScreen from 'expo-splash-screen';
+
+// Your main App component
+import Main from './components/Main';
+
+const App = () => {
+  const [isLoadingComplete, setLoadingComplete] = useState(false);
+
+  const appLoading = async () => {
+    try {
+      // keep showing the SplashScreen 
+      await SplashScreen.preventAutoHideAsync();
+      // do any other app initialization like loading resources etc.
+      loadResourcesAsync();
+    } catch (error) {
+      // handle error
+    } finally {
+      setLoadingComplete(true);
+      // hide the SplashScreen after all initialization is done.
+      await SplashScreen.hideAsync();
+    }
+  };
+
+  useEffect(() => {
+    appLoading();
+  }, []);
+
+  return isLoadingComplete ? <Main /> : null;
+};
+
+const loadResourcesAsync = async () => {
+  // Cache images, fonts etc.
+}
+```
```

Comment:
Sure, you can abstract it into a custom hook. I'm fine with it. And it might be nice to even offer one in the lib once. 

However, I didn't want to make the example too difficult. Could you also add the comments I had in my example to yours? I like it when example code also explains me why things are there. Then I directly know why the example was built that way.

Reply:
@andrekovac, feel free to adjust my suggestion according to your ideas 😉
Once you're done with it, let's merge it 🎉 
Regarding:
> And it might be nice to even offer one in the lib once.
if you are willing to provide PR with one I'll be happy to merge it 😉